### PR TITLE
Demo desktop

### PR DESCRIPTION
**Pull request by danielcaze**
This PR will add the ability to build the application and run it on your desktop.

image
image

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const { app, BrowserWindow } = require("electron");
+const path = require("path");
+
+function createWindow() {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
```
Comment (Sanscripter) [inline main.js:7]: can this be configured to be Fullscreen?
can this be configured internally in the app?

File: package.json [hunk 2]
```diff
@@ -1,16 +1,60 @@
...
     "storybook": "ng run trading-grounds:storybook",
-    "build-storybook": "ng run trading-grounds:build-storybook"
+    "build-storybook": "ng run trading-grounds:build-storybook",
+    "electron": "npm run build && electron .",
+    "electron-dev": "concurrently -k \"ng serve\" \"set NODE_ENV=development && electron .\"",
+    "electron-build": "npm run build && electron-builder"
   },
   "private": true,
+  "build": {
+    "appId": "com.trade_runner.id",
```
Comment (Sanscripter) [inline package.json:25]: what is this used for?

**Comment by Sanscripter** [comment]
How do we change shortcut logo? Please create Trello task to track that.

File: package.json (hunk 2 above)
Comment (danielcaze) [reply package.json:25]: The appId in our build config is like an unique ID for our app. It's used during packaging and distribution to help the OS recognize our app as distinct from others. This ensures things like installations, updates, and integrations (like notifications or opening certain file types) work smoothly without any conflicts.

File: main.js (hunk 1 above)
Comment (danielcaze) [reply main.js:7]: Yes it can, just remove width and height and add fullscreen as a boolean.

```suggestion
    fullscreen: true,
```

**Comment by danielcaze** [comment]
> How do we change shortcut logo? Please create Trello task to track that.

To change the app icon you need to add "icon" property on each platform build, on the package.json.

package.json

```json
{
    "win": {
      "target": [
        "nsis",
        "zip"
      ],
      "icon": "icon.ico"
    },
    "linux": {
      "target": [
        "AppImage",
        "deb"
      ],
      "category": "games",
      "icon": "icon.ico"
    },
    "mac": {
      "target": "dmg",
      "category": "games",
      "icon": "icon.ico"
    }
}
```

As for the window icon (the one on the first image, followed by the game name), you need to set on the electron window opening

like this:
```js
  const win = new BrowserWindow({
    width: 800,
    height: 600,
    icon: path.join(__dirname, "path/to/icon.png"),
    webPreferences: {
      nodeIntegration: true,
      contextIsolation: false,
    },
  });
```

File: main.js (hunk 1 above)
Comment (Sanscripter) [reply main.js:7]: More concerned about having the option to tinker with that in configuration than with actually making it fullscreen atm. Open a ticket for that and we can approve this.

**Comment by Sanscripter** [comment]
Feel free to add any of the trader mugshots as an icon right now and then merge. Make sure read explains how to build in desktop mode.

**Comment by danielcaze** [comment]
> Feel free to add any of the trader mugshots as an icon right now and then merge. Make sure readme explains how to build in desktop mode.

I tried using the tv gods but it didn't have background, so I used Corey`s

image

**Review by Sanscripter — APPROVED** [review]

**Pull request merged**